pandas/tests/base/test_ops.py

```diff
@@ -137,227 +137,233 @@ def setup_method(self, method):
-                    msg = "Invalid comparison"
-                    with pytest.raises(TypeError, match=msg):
-                        None > o
-                    with pytest.raises(TypeError, match=msg):
-                        o > None
-                else:
-                    result = None > o
-                    assert not result.iat[0]
-                    assert not result.iat[1]
+        o[0] = np.nan
+
+        # noinspection PyComparisonWithNone
+        result = o == None  # noqa
+        assert not result.iat[0]
+        assert not result.iat[1]
+
+        # noinspection PyComparisonWithNone
+        result = o != None  # noqa
+        assert result.iat[0]
+        assert result.iat[1]
+
+        result = None == o  # noqa
+        assert not result.iat[0]
+        assert not result.iat[1]
+
+        result = None != o  # noqa
+        assert result.iat[0]
+        assert result.iat[1]
+
+        if is_datetime64_dtype(o) or is_datetime64tz_dtype(o):
+            # Following DatetimeIndex (and Timestamp) convention,
+            # inequality comparisons with Series[datetime64] raise
+            msg = "Invalid comparison"
+            with pytest.raises(TypeError, match=msg):
+                None > o
+            with pytest.raises(TypeError, match=msg):
+                o > None
+        else:
+            result = None > o
+            assert not result.iat[0]
+            assert not result.iat[1]
 
-                    result = o < None
-                    assert not result.iat[0]
-                    assert not result.iat[1]
+            result = o < None
+            assert not result.iat[0]
+            assert not result.iat[1]
 
-    def test_ndarray_compat_properties(self):
+    def test_ndarray_compat_properties(self, index_or_series_obj):
+        o = index_or_series_obj
 
-        for o in self.objs:
-            # Check that we work.
-            for p in ["shape", "dtype", "T", "nbytes"]:
-                assert getattr(o, p, None) is not None
+        # Check that we work.
+        for p in ["shape", "dtype", "T", "nbytes"]:
+            assert getattr(o, p, None) is not None
 
-            # deprecated properties
-            for p in ["flags", "strides", "itemsize", "base", "data"]:
-                assert not hasattr(o, p)
+        # deprecated properties
+        for p in ["flags", "strides", "itemsize", "base", "data"]:
+            assert not hasattr(o, p)
 
-            msg = "can only convert an array of size 1 to a Python scalar"
-            with pytest.raises(ValueError, match=msg):
-                o.item()  # len > 1
+        msg = "can only convert an array of size 1 to a Python scalar"
+        with pytest.raises(ValueError, match=msg):
+            o.item()  # len > 1
 
-            assert o.ndim == 1
-            assert o.size == len(o)
+        assert o.ndim == 1
+        assert o.size == len(o)
 
         assert Index([1]).item() == 1
         assert Series([1]).item() == 1
```

Comment:
these last two lines look like they belong somewhere else

Reply:
Indeed. I wouldn't want to address it in this PR yet though